statsd/worker.go

```diff
@@ -3,23 +3,31 @@ package statsd
 import (
 	"math/rand"
 	"sync"
+	"time"
 )
 
 type worker struct {
 	pool   *bufferPool
 	buffer *statsdBuffer
 	sender *sender
+	random *rand.Rand
 	sync.Mutex
 
 	inputMetrics chan metric
 	stop         chan struct{}
 }
 
 func newWorker(pool *bufferPool, sender *sender) *worker {
+	// Note that calling time.Now().UnixNano() repeatedly quickly may return
+	// very similar values. That's fine for seeding the worker-specific random
+	// source because we just need an evenly distributed stream of float values.
+	// Do not use this random source for cryptographic randomness.
+	random := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
Do you mind adding a comment quickly explaining why each worker has its own random ?

Reply:
Sure, will add a comment.